smart-contracts/smart-contract/tests/main-contract_test.ts

```diff
@@ -48,6 +51,134 @@ const err_not_proposed_notifier = '(err u124)';
+    assertEquals(block.receipts.length, 2);
+    assertEquals(block.height, 653);
+
+    block = chain.mineBlock([
+      Tx.contractCall(CONTRACT_NAME, ASK_TO_JOIN, [types.ascii(accounts.get(`wallet_${300}`)!.address)], accounts.get(`wallet_${300}`)!.address),
+      Tx.contractCall(CONTRACT_NAME, GET_DATA_WAITING_MINER, [types.list([types.principal(accounts.get(`wallet_${300}`)!.address)])], deployer.address),
+    ]);
+
+    for (let i = 91; i <= 299; i++) {
+      const miner = accounts.get(`wallet_${300}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, VOTE_POSITIVE_JOIN, [types.principal(miner.address)], accounts.get(`wallet_${i}`)!.address),
+      ]);
+    }
+
+    for (let i = 1; i <= 90; i++) {
+      const miner = accounts.get(`wallet_${300}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, VOTE_NEGATIVE_JOIN, [types.principal(miner.address)], accounts.get(`wallet_${i}`)!.address),
+      ]);
+    }
+
+    block = chain.mineBlock([
+      Tx.contractCall(CONTRACT_NAME, GET_DATA_WAITING_MINER, [types.list([types.principal(accounts.get(`wallet_${300}`)!.address)])], deployer.address),
+    ]);
+  },
+});
+
+Clarinet.test({
+  name: 'Get All Data Removals 300 Miners',
+  async fn(chain: Chain, accounts: Map<string, Account>) {
+    const deployer = accounts.get('deployer')!;
+    let waiting_list = [];
+    let block = chain.mineBlock([]);
+    assertEquals(block.receipts.length, 0);
+    assertEquals(block.height, 2);
+
+    for (let i = 1; i <= 299; i++) {
+      const miner = accounts.get(`wallet_${i}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, ASK_TO_JOIN, [types.ascii(miner.address)], miner.address),
+      ]);
+      if (i == 1) waiting_list.push(`${miner.address}`);
+      else waiting_list.push(` ${miner.address}`);
+    }
+    block = chain.mineBlock([Tx.contractCall(CONTRACT_NAME, GET_WAITING_LIST, [], deployer.address)]);
+    assertEquals(block.receipts[0].result, `[${waiting_list}]`);
+    assertEquals(block.receipts.length, 1);
+    assertEquals(block.height, 302);
+
+    for (let i = 1; i <= 299; i++) {
+      const miner = accounts.get(`wallet_${i}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, VOTE_POSITIVE_JOIN, [types.principal(miner.address)], deployer.address),
+        Tx.contractCall(CONTRACT_NAME, TRY_ENTER_POOL, [], miner.address),
+      ]);
+    }
+
+    block = chain.mineBlock([Tx.contractCall(CONTRACT_NAME, GET_PENDING_LIST, [], deployer.address)]);
+    assertEquals(block.receipts[0].result, `[${waiting_list}]`);
+    assertEquals(block.receipts.length, 1);
+    assertEquals(block.height, 602);
+
+    for (let i = 1; i <= 50; i++) {
+      block = chain.mineBlock([]);
+    }
+
+    block = chain.mineBlock([
+      Tx.contractCall(CONTRACT_NAME, ADD_PENDING_MINERS, [], deployer.address),
+      Tx.contractCall(CONTRACT_NAME, GET_MINERS_LIST, [], deployer.address),
```

Comment:
> @BowTiedJohn can we add another assertEquals to check the length of the miners list variable?
also add assert here for this

Reply:
> We don't keep the pending list locally, but I am doing the checks for it through waiting list (since everyone in waiting list is being transferred to pending). Also, on the next line, we check that the pending list is equal to our local waiting list, which makes the requested assert useless, because we are directly checking for address list to match SC call, rather than checking for the list length.

Also checking for '[deployer, waiting_list]' on the following line, since all pending miners should be in pool, with the deployer already being there first.